src/field.rs

```diff
@@ -77,6 +77,13 @@ pub trait Field:
 
         term
     }
+
+    /// Blanket implementation to represent the instance of this trait as 16 byte integer.
+    /// Uses the fact that such conversion already exists via `Self` -> `Self::Integer` -> `Into<u128>`
+    fn as_u128(&self) -> u128 {
+        let int: Self::Integer = (*self).into();
```

Comment:
Why not

```suggestion
        let int: Self::Integer = self.into();
```

Or for the entire thing:

```rust
<self as Into<Self::Integer>>.into().into()

Reply:
I believe compiler is getting confused because of `&self` which implies a reference to self while `Into` is defined for `Field` and not for `&Field`. I am not sure why it does not attempt to dereference it (it is `Clone` and `Copy`) so that was the only way I could convince it that conversion works

wrt to your second suggestion to convert it to a trait object (Into is a trait) - I don't think that would work because `Into` is not an object-safe trait

```rust
error[E0038]: the trait `Into` cannot be made into an object
  --> src/field.rs:86:18
   |
86 |         (self as &dyn Into<Self::Integer>).into().into()
   |                  ^^^^^^^^^^^^^^^^^^^^^^^^ `Into` cannot be made into an object
```